### ISSUE TITLE
Error Message for Non Admin users

### ISSUE THREAD

**Issue by suture** [A5hleyRich/BuddyPress-Admin-Only-Profile-Fields#13]
I set "Visibility" to "Admin" for a field and check "Enforce field visibility". But when I login as a non-admin user and try to edit my Extend profile I get the following error message.

> There was a problem updating some of your profile information. Please try again.

It's saving the data, but generating the above error. If I disable the plugin then the user can edit their profile without the above error message.

**Comment by suture** [comment]
In the docs it's states 

> In the WordPress admin area, go to _Users > Profile Fields_ and click _Edit_ on the desired profile field. > Under the _Default Visibility_ panel select _Hidden_ as the value and click _Save_.

However I'm not seeing this value "Hidden". 

![screen shot 2015-11-24 at 4 06 33 pm](https://cloud.githubusercontent.com/assets/90376/11360168/619b9dbc-92c5-11e5-92d6-3451885e1bb8.png)

**Comment by garretthyder** [comment]
Hi @suture can you provide the versions for PHP, Wordpress, Buddypress and the plugin that you're using.
And what additional plugins are currently installed.
Also if you can provide access to you instance can investigate further. I just installed a fresh instance without issue so am assuming it's a conflict.
Cheers

**Comment by suture** [comment]
Hi @garrett-eclipse thanks for getting back to me.

PHP 5.6
Wordpress 4.3.1 
BuddyPress 2.4.0
BuddyPress Admin Only Profile Fields 1.2 downloaded from https://wordpress.org/plugins/buddypress-admin-only-profile-fields/

I've deactivated all other plugins and tested with Twenty Fifteen Version: 1.3, but still getting the same error.
The staging server is located here http://viscopy.suture.net/

**Comment by garretthyder** [comment]
Thanks @suture, I set-up an instance but sadly had no luck in reproducing. Any chance I can get an admin account on your staging server? You can email it to me at nightbook [dot] g [dot] hyder [at] gmail [dot] com
Cheers

**Comment by suture** [comment]
Hi Garrett, I've created an admin account for you. The new account email with login details should be coming through to you now. If any problems please let me know. The error only occurs when a non-admin user (subscriber etc) tries to update their Extended Profile. 

Many thanks again for your help.

**Comment by suture** [comment]
Hi Garrett, please feel free to disable plugins, switch themes etc, if needed. It's the staging version, so not really possible to do any damage. 

The same error is happening though on my local version and the live production versions.

**Comment by suture** [comment]
I worked out how to reproduce the error.
1. Select "Create New Field Group" and create a new group (for example: "Admin Options")
2. Select "Add New Field" and add a single field to the new group.
3. Mark that field as "Hidden" or "Admin Only" and don't add anymore fields.
4. Login as a Subscriber and go to "Extended Profile" you will see an empty Group called "Admin Options"
5. Select "Update Profile" and then you will get the error: "There was a problem updating some of your profile information. Please try again."
6. If you login back in as an admin user and add another field to the "Admin Options" group, but don't set it as "Hidden" the problem goes away.

I guess we could close this ticket. It doesn't make sense to have empty groups, so it's probably not an issue in most cases.

**Comment by garretthyder** [comment]
Thanks @suture, sorry wasn't able to delve on the weekend had my daughters bday. 

I was able to login there, and also on my own instance following your steps was able to recreate the issue.  Sorry hadn't realized you were on the Extended Profile from the backend there. 

Seems if a group only consists of admin fields it will fail to save. I'll delve into this further tonight and hopefully have a patch. I can see some cases where people would want their admin fields all in a group so as to keep them together there so we'll want to resolve this.

Also I did set-up a garrett-subscriber on your instance for testing, when I've finished and resolved will let you know and you can remove my users from your system.

Cheers

**Comment by garretthyder** [comment]
Sidenote: Appears I get the following errors;
[30-Nov-2015 21:55:03 UTC] PHP Notice:  Trying to get property of non-object in /home/dev/public_html/buddypress/wp-content/plugins/buddypress/bp-xprofile/bp-xprofile-template.php on line 738
[30-Nov-2015 21:55:03 UTC] PHP Notice:  Trying to get property of non-object in /home/dev/public_html/buddypress/wp-content/plugins/buddypress/bp-xprofile/bp-xprofile-template.php on line 630

**Comment by suture** [comment]
Hi @garrett-eclipse 
Many thanks again for following this up. That sounds good.

**Comment by garretthyder** [comment]
Hi @suture, 

I was able to isolate the issue in Buddypress and have opened a track for a proper resolution;
https://buddypress.trac.wordpress.org/ticket/6757#ticket

And am looking for a temporary workaround in the meantime.

Cheers

**Comment by suture** [comment]
Hi Garrett, that is some excellent news. Also I like the idea of not displaying groups that don't have any available fields.

**Comment by garretthyder** [comment]
Hi @suture, 

I've found a workaround and set-up a branch for this issue;
https://github.com/A5hleyRich/buddypress-admin-only-profile-fields/tree/issue13

I don't have FTP or the ability to edit files via your WP installation so please download the zip from the branch and update the plugin to test.

Thank you

**Comment by suture** [comment]
Hi @garrett-eclipse That's working perfectly. No error message and the admin group is nicely hidden.
Many thanks

**Comment by garretthyder** [comment]
Perfect, thanks for testing @suture.

@A5hleyRich when you can mind rolling a 1.2.1 patch. Thanks

**Comment by suture** [comment]
Hi @garrett-eclipse one small issue. Don't shoot me :) 

If an Admin User edits their own extended profile, they currently can't see the hidden "Admin" fields. But they can see the hidden fields when editing other users. 

So it works correctly when editing other users, but not when editing their own profile. 

It's not a big issue. I can live with it. But maybe worth mentioning.

**Comment by garretthyder** [comment]
Hi @suture,

No worries, we're sort it out, I just logged in as my admin and went to the Extended Profile on the backend for that users. I did find the Admin Options group and field there for editing so not sure I see your issue.

What user(s) specifically are experiencing this, they may have a user level lower than administrator and so would have their admin fields hidden. It's only the Administrator role which will allow access to the field on their own and others profiles.

Hoping it's simply a roles issue.

Cheers
